src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseResource.java

```diff
@@ -531,6 +531,26 @@ public ResponseEntity<ProgrammingExercise> getProgrammingExerciseWithTemplateAnd
         return ResponseEntity.ok(programmingExercise);
     }
 
+    /**
+     * GET /programming-exercises/:exerciseId/with-auxiliary-repository/:auxiliaryRepositoryId
+     *
+     * @param exerciseId            the id of the programmingExercise to retrieve
+     * @param auxiliaryRepositoryId the id of the auxiliary repository of the exercise
+     * @return the ResponseEntity with status 200 (OK) and the programming exercise with template and solution participation, or with status 404 (Not Found)
+     */
+    @GetMapping("programming-exercises/{exerciseId}/with-auxiliary-repository/{auxiliaryRepositoryId}")
+    @EnforceAtLeastTutorInExercise
+    public ResponseEntity<ProgrammingExercise> getProgrammingExerciseWithTemplateAndSolutionParticipation(@PathVariable long exerciseId, @PathVariable long auxiliaryRepositoryId) {
+
+        log.debug("REST request to get programming exercise with template and solution participation : {}", exerciseId);
```

Comment:
Could you update the method name and log to reflect the endpoint? I feel like it should mention the auxiliary repository